packages/driver/src/dom/elements/find.ts

```diff
@@ -224,55 +223,91 @@ export const getElements = ($el) => {
+ * `getContainsSelector` below.
+ */
 
-  const escapedText = escapeQuotes(
-    escapeBackslashes(text),
-  )
+// Example:
+// button:cy-contains("Login")
+$.expr[':']['cy-contains'] = $.expr.createPseudo((text) => {
+  text = JSON.parse(`"${ text }"`)
 
-  // they may have written the filter as
-  // comma separated dom els, so we want to search all
-  // https://github.com/cypress-io/cypress/issues/2407
-  const filters = filter.trim().split(',')
+  return function (elem) {
+    let testText = normalizeWhitespaces(elem)
 
-  let cyContainsSelector
+    return testText.includes(text)
+  }
+})
 
-  if (_.isRegExp(text)) {
-    if (options.matchCase === false && !text.flags.includes('i')) {
-      text = new RegExp(text.source, text.flags + 'i') // eslint-disable-line prefer-template
-    }
+// Example:
+// .login-button:cy-contains-insensitive("login")
+$.expr[':']['cy-contains-insensitive'] = $.expr.createPseudo((text) => {
+  text = JSON.parse(`"${ text }"`)
 
-    // taken from jquery's normal contains method
-    cyContainsSelector = function (elem) {
-      if (elem.type === 'submit' && elem.tagName === 'INPUT') {
-        return text.test(elem.value)
-      }
+  return function (elem) {
+    let testText = normalizeWhitespaces(elem)
 
-      const testText = normalizeWhitespaces(elem)
+    testText = testText.toLowerCase()
+    text = text.toLowerCase()
 
-      return text.test(testText)
-    }
-  } else if (_.isString(text)) {
-    cyContainsSelector = function (elem) {
-      let testText = normalizeWhitespaces(elem)
+    return testText.includes(text)
+  }
+})
+
+function isSubmit (elem: Element): elem is HTMLInputElement {
+  return elem.tagName === 'INPUT' && (elem as HTMLInputElement).type === 'submit'
+}
 
-      if (!options.matchCase) {
-        testText = testText.toLowerCase()
-        text = text.toLowerCase()
-      }
+// Example:
+// #login>li:first:cy-contains-regex('/asdf 1/i')
+$.expr[':']['cy-contains-regex'] = $.expr.createPseudo((text) => {
+  const lastSlash = text.lastIndexOf('/')
+  const regex = new RegExp(text.slice(1, lastSlash), text.slice(lastSlash + 1))
 
-      return testText.includes(text)
+  // taken from jquery's normal contains method
+  return function (elem) {
+    if (isSubmit(elem)) {
+      return regex.test(elem.value)
     }
-  } else {
-    cyContainsSelector = $expr.contains
+
+    const testText = normalizeWhitespaces(elem)
+
+    return regex.test(testText)
+  }
+})
+
+export const getContainsSelector = (text, filter = '', options: {
+  matchCase?: boolean
+} = {}) => {
+  if (_.isRegExp(text) && options.matchCase === false && !text.flags.includes('i')) {
+    text = new RegExp(text.source, text.flags + 'i') // eslint-disable-line prefer-template
   }
 
-  // we set the `cy-contains` jquery selector which will only be used
-  // in the context of cy.contains(...) command and selector playground.
-  $expr['cy-contains'] = cyContainsSelector
```

Comment:
Yep. In develop this is overridden each time `.contains()` retries, with cyContainsSelector holding internal state (rather than letting jquery parse the args and pass it in).